### PR TITLE
fix issue #148

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

**Comment by SummerSun** [comment]
fix issue #148

File: src/TomcatSeverTreeProvider.ts [hunk 1]
```diff
@@ -39,11 +39,22 @@ export class TomcatSeverTreeProvider implements vscode.TreeDataProvider<TreeItem
...
-                const wars: string[] = await fse.readdir(webapps);
-                return wars.map((w: string) => {
+                let wars: string[] = await fse.readdir(webapps);
+                let temp: fse.Stats;
+                // show war packages with no extension if there is one
+                // and no need to show war packages if its unzipped folder exists
+                wars = wars.filter((w: string) => {
                     if (w.toUpperCase() !== 'ROOT') {
-                        return new WarPackage(w, server.getName(), iconPath, path.join(webapps, w));
+                        temp = fse.statSync(path.join(webapps, w));
```
Comment (jdneo) [inline src/TomcatSeverTreeProvider.ts]: stat has an async version: `export function stat(path: string | Buffer): Promise<Stats>;`

File: src/TomcatSeverTreeProvider.ts [hunk 2]
```diff
@@ -39,11 +39,22 @@ export class TomcatSeverTreeProvider implements vscode.TreeDataProvider<TreeItem
...
+                        if (temp.isDirectory() || (temp.isFile() && w.endsWith('.war'))) {
```
Comment (jdneo) [inline src/TomcatSeverTreeProvider.ts]: we can use `path.extname()`

File: src/TomcatSeverTreeProvider.ts [hunk 3]
```diff
@@ -39,21 +41,23 @@ export class TomcatSeverTreeProvider implements vscode.TreeDataProvider<TreeItem
...
-                wars = wars.filter((w: string) => {
+                const promises: Promise<void>[] = (await fse.readdir(webapps)).map(async (w: string) => {
                     if (w.toUpperCase() !== 'ROOT') {
-                        temp = fse.statSync(path.join(webapps, w));
-                        if (temp.isDirectory() || (temp.isFile() && w.endsWith('.war'))) {
-                            return w;
+                        temp = await fse.stat(path.join(webapps, w));
+                        fileExtension = path.extname(path.join(webapps, w));
+                        if (temp.isDirectory() || (temp.isFile() && fileExtension === Constants.WAR_FILE_EXTENSION)) {
+                            wars.push(fileExtension === Constants.WAR_FILE_EXTENSION ? w.substring(0, w.indexOf(fileExtension)) : w);
```
Comment (jdneo) [inline src/TomcatSeverTreeProvider.ts]: Shall we replace ` w.substring(0, w.indexOf(fileExtension)` with `path.basename(w, fileExtension)`?

Comment (SummerSun) [reply src/TomcatSeverTreeProvider.ts]: yes, this looks nice~

**Pull request merged**